### PR TITLE
feat: update qdrant version to 0.10.1

### PR DESCRIPTION
**Pull request by JoanFM**
Goals:

- [x] update qdrant version to 0.10.1
- [x] update some of the tests (add grpc tests for qdrant)
- [ ] update docs

### CODE REVIEW

File: docarray/array/storage/qdrant/backend.py [hunk 1]
```diff
@@ -37,6 +39,10 @@ class QdrantConfig:
     collection_name: Optional[str] = None
     host: Optional[str] = field(default="localhost")
     port: Optional[int] = field(default=6333)
+    grpc_port: Optional[int] = field(default=6334)
+    prefer_grpc: Optional[bool] = field(default=True)
```
Comment (alaeddine-13) [inline docarray/array/storage/qdrant/backend.py:43]: follow db default behaviour:
```suggestion
    grpc_port: Optional[int] = None
    prefer_grpc: Optional[bool] = None
```

File: docarray/array/storage/qdrant/backend.py [hunk 2]
```diff
@@ -37,6 +39,10 @@ class QdrantConfig:
...
+    api_key: Optional[str] = field(default=None)
+    https: Optional[bool] = field(default=None)
```
Comment (alaeddine-13) [inline docarray/array/storage/qdrant/backend.py:45]:
```suggestion
    api_key: Optional[str] = None
    https: Optional[bool] = None
```

**Review by alaeddine-13 — COMMENTED** [review]
can we add some grpc tests for qdrant ? just add an extra case to some of the existing tests, instead of just
```python
        (DocumentArrayQdrant, QdrantConfig(n_dim=128)),
```
we do:
```python
        (DocumentArrayQdrant, QdrantConfig(n_dim=128)),
        (DocumentArrayQdrant, QdrantConfig(n_dim=128, prefer_grpc=True)),
```

**Comment by JoanFM** [comment]
Please @AnneYang720 reopen this PR under ur name

**Pull request closed without merging**